### PR TITLE
Ensure we really wait until operation is complete.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

File: searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp [hunk 1]
```diff
@@ -594,7 +594,8 @@ struct FixtureBase
     void putAndWait(const DocumentContext &docCtx) {
         FeedTokenContext token(_tracer);
         PutOperation op(docCtx.bid, docCtx.ts, docCtx.doc);
-        runInMaster([&] () { performPut(token.ft, op); });
+        runInMaster([this, ft=std::move(token.ft), &op] () { performPut(std::move(ft), op); });
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp:597]: Consider adding mutable specifier to the lambda expression.

File: searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp [hunk 2]
```diff
@@ -606,7 +607,8 @@ struct FixtureBase
     void updateAndWait(const DocumentContext &docCtx) {
         FeedTokenContext token(_tracer);
         UpdateOperation op(docCtx.bid, docCtx.ts, docCtx.upd);
-        runInMaster([&] () { performUpdate(token.ft, op); });
+        runInMaster([this, ft=std::move(token.ft), &op] () { performUpdate(std::move(ft), op); });
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp:610]: Consider adding mutable specifier to the lambda expression.

File: searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp [hunk 3]
```diff
@@ -620,7 +622,8 @@ struct FixtureBase
     void removeAndWait(const DocumentContext &docCtx) {
         FeedTokenContext token(_tracer);
         RemoveOperationWithDocId op(docCtx.bid, docCtx.ts, docCtx.doc->getId());
-        runInMaster([&] () { performRemove(token.ft, op); });
+        runInMaster([this, ft=std::move(token.ft), &op] () { performRemove(std::move(ft), op); });
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp:625]: Consider adding mutable specifier to the lambda expression.

**Review by toregge — CHANGES_REQUESTED** [review]

File: searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp]: Fixed

File: searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp (hunk 3 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/feedview/feedview_test.cpp]: Fixed

**Review by toregge — COMMENTED** [review]
👍

**Pull request merged**